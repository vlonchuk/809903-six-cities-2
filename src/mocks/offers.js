const offers = [
  {
    id: `prop-1`,
    city: `Amsterdam`,
    caption: `Beautiful & luxurious apartment at great location`,
    imgSrc: `img/apartment-01.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 120,
    priceText: `night`,
    rating: 50,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    },
  },
  {
    id: `prop-2`,
    city: `Amsterdam`,
    caption: `Wood and stone place`,
    imgSrc: `img/room.jpg`,
    type: `Private room`,
    priceCurrency: `€`,
    priceValue: 80,
    priceText: `night`,
    rating: 80,
    coor: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
  },
  {
    id: `prop-3`,
    city: `Amsterdam`,
    caption: `Canal View Prinsengracht`,
    imgSrc: `img/apartment-02.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 132,
    priceText: `night`,
    rating: 100,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
  },
  {
    id: `prop-4`,
    city: `Amsterdam`,
    caption: `Nice, cozy, warm big bed apartment`,
    imgSrc: `img/apartment-03.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 180,
    priceText: `night`,
    rating: 65,
    coor: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198
    },
  },
  {
    id: `prop-5`,
    city: `Paris`,
    caption: `Canal View Prinsengracht`,
    imgSrc: `img/apartment-02.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 132,
    priceText: `night`,
    rating: 90,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
  },
  {
    id: `prop-6`,
    city: `Paris`,
    caption: `Nice, cozy, warm big bed apartment`,
    imgSrc: `img/apartment-03.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 180,
    priceText: `night`,
    rating: 40,
    coor: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198
    },
  },
];

export const getPropertiesByCity = (city) => {
  return offers.filter((el) => el.city === city)
               .slice();
};

export default offers;
