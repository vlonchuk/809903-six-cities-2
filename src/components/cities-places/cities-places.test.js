import React from 'react';
import renderer from 'react-test-renderer';
import CitiesPlaces from './cities-places';
import SortType from './../../consts/sort-type.js';
import offers from './../../mocks/offers.js';

jest.mock(`./../map/map.jsx`, () => jest.fn().mockReturnValue(null));

it(`CitiesPlaces correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<CitiesPlaces
      offers={offers} city={offers[0].city.name} properties={offers} loadOffers={jest.fn()}
      sortOptions={[SortType.POPULAR]}
      sortActiveOption={SortType.POPULAR}
      sortOpened={false}
      onSortArrowClick={jest.fn()}
      onSortOptionClick={jest.fn()}
      onPlaceCardMouseEnter={jest.fn()}
      onPlaceCardMouseLeave={jest.fn()}
      selectedCity={``}
      onClick={jest.fn()}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
