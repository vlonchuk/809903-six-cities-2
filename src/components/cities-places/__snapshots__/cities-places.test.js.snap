// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesPlaces correctly renders after relaunch 1`] = `
<div
  className="cities__places-container container"
>
  <section
    className="cities__places places"
  >
    <h2
      className="visually-hidden"
    >
      Places
    </h2>
    <b
      className="places__found"
    >
      1
       places to stay in 
      Amsterdam
    </b>
    <form
      action="#"
      className="places__sorting"
      method="get"
    >
      <span
        className="places__sorting-caption"
      >
        Sort by
      </span>
      <span
        className="places__sorting-type"
        onClick={[Function]}
        tabIndex="0"
      >
        Â 
        Popular
        <svg
          className="places__sorting-arrow"
          height={4}
          width={7}
        >
          <use
            xlinkHref="#icon-arrow-select"
          />
        </svg>
      </span>
      <ul
        className="places__options places__options--custom"
      >
        <li
          className="places__option places__option--active"
          onClick={[Function]}
          tabIndex="0"
        >
          Popular
        </li>
      </ul>
    </form>
  </section>
  <div
    className="cities__right-section"
  />
</div>
`;
